### PR TITLE
installation: RabbitMQ and anyjson requirements

### PR DESCRIPTION
**Pull request by Dziolas**
* Adds missing anyjson and amqp packages.

Signed-off-by: Wojciech Ziolek <wojciech.ziolek@cern.ch>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -55,6 +55,8 @@ def run(self):
 
 install_requires = [
     "alembic>=0.6.6",
+    "amqp>=1.4.5,<2.0",
+    "anyjson>=0.3.3",
```
Comment (jirikuncar) [inline setup.py]: where do you use `anyjson`?

File: setup.py [hunk 2]
```diff
@@ -55,6 +55,8 @@ def run(self):
 
 install_requires = [
     "alembic>=0.6.6",
+    "amqp>=1.4.5,<2.0",
```
Comment (jirikuncar) [inline setup.py]: `amqp` should be optional extra `"rabbitmq": "amqp>=1.4.5,<2.0",`. Why do you need to bind the maximum version?

**Comment by jirikuncar** [comment]
please amend component name: `global` -> `installation`

File: setup.py (hunk 1 above)
Comment (Dziolas) [inline setup.py]: I was installing pu on my local machine and when I was trying to run `inveniomanage bower` I got error that anyjson is missing.

File: setup.py (hunk 2 above)
Comment (Dziolas) [inline setup.py]: I took version from the messages I got running `inveniomanage bower`.

File: setup.py (hunk 1 above)
Comment (jirikuncar) [inline setup.py]: It might be internal dependency. Have you tried reinstalling/upgrading Invenio package?

File: setup.py (hunk 2 above)
Comment (Dziolas) [inline setup.py]: Traceback (most recent call last):
  File "/home/wojciech/.virtualenvs/scoap3-next/bin/inveniomanage", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/home/wojciech/.virtualenvs/scoap3-next/local/lib/python2.7/site-packages/pkg_resources.py", line 2716, in <module>
    working_set.require(__requires__)
  File "/home/wojciech/.virtualenvs/scoap3-next/local/lib/python2.7/site-packages/pkg_resources.py", line 685, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/wojciech/.virtualenvs/scoap3-next/local/lib/python2.7/site-packages/pkg_resources.py", line 588, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: amqp>=1.4.5,<2.0

Comment (jirikuncar) [inline setup.py]: please move it to `setup.py:extras_requires["rabbitmq"] = "amqp>=1.4.5"`

File: setup.py (hunk 1 above)
Comment (Dziolas) [inline setup.py]: Even with the newest PU and after reinstall I get this:

 inveniomanage runserverTraceback (most recent call last):
  File "/home/wojciech/.virtualenvs/scoap3-next/bin/inveniomanage", line 5, in <module>
    from pkg_resources import load_entry_point
  File "/home/wojciech/.virtualenvs/scoap3-next/local/lib/python2.7/site-packages/pkg_resources.py", line 2716, in <module>
    working_set.require(__requires__)
  File "/home/wojciech/.virtualenvs/scoap3-next/local/lib/python2.7/site-packages/pkg_resources.py", line 685, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/home/wojciech/.virtualenvs/scoap3-next/local/lib/python2.7/site-packages/pkg_resources.py", line 588, in resolve
    raise DistributionNotFound(req)
pkg_resources.DistributionNotFound: anyjson>=0.3.3

File: setup.py [hunk 3]
```diff
@@ -55,6 +55,7 @@ def run(self):
 
 install_requires = [
     "alembic>=0.6.6",
+    "anyjson>=0.3.3",
```
Comment (jirikuncar) [inline setup.py]: @invenio-developers does anybody have a problem with missing `anyjson` package?

Comment (jirikuncar) [inline setup.py]: @Dziolas because nooneelse reported similar problem, please remove `anyjson` from the `setup.py`.

**Comment by jirikuncar** [comment]
@Dziolas you can add me as reviewer. Then please rebase the branch and ping me on hangouts. Thanks

**Comment by jirikuncar** [comment]
@Dziolas please ping me when you are done before you rebase. Thanks

**Comment by jirikuncar** [comment]
@Dziolas can you quickly amend the commit according to https://github.com/inveniosoftware/invenio/pull/2765#discussion_r25053441 and rebase. Thanks

--
cc @jalavik @jmartinm

**Comment by Dziolas** [comment]
Sorry, I totally forgot about it. Will be done tomorrow morning.

2015-03-03 17:45 GMT+01:00 Jiri Kuncar <notifications@github.com>:

>  @Dziolas <https://github.com/Dziolas> can you quickly amend the commit
> according to #2765 (comment)
> <https://github.com/inveniosoftware/invenio/pull/2765#discussion_r25053441>
> and rebase. Thanks
>
> cc @jalavik <https://github.com/jalavik> @jmartinm
> <https://github.com/jmartinm>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/inveniosoftware/invenio/pull/2765#issuecomment-76983141>
> .
>



-- 
Wojciech Ziółek

*SCOAP3 Repository Manager*
*CERN, Switzerland*

**Pull request merged**